### PR TITLE
fix: use ENCRYPTION_KEY in single use surveys

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/actions.ts [hunk 1]
```diff
@@ -2,6 +2,5 @@
 import { generateSurveySingleUseId } from "@/lib/singleUseSurveys";
 
 export async function generateSingleUseIdAction(isEncrypted: boolean): Promise<string> {
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/actions.ts:4]: if you currently have this open, please add auth :-)

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/page.tsx [hunk 2]
```diff
@@ -17,7 +17,8 @@ export default async function SurveysEditPage({ params }) {
     getActionClasses(params.environmentId),
     getAttributeClasses(params.environmentId),
   ]);
-  const isEncryptionKeySet = !!FORMBRICKS_ENCRYPTION_KEY;
+
+  const isEncryptionKeySet = !!ENCRYPTION_KEY;
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/page.tsx:21]: I think we can skip this as it's not possible to not have the ENCRYPTION_KEY set. So please also remove the disabled state of the encryption toggle together with the tooltip :-)

File: apps/web/lib/singleUseSurveys.ts [hunk 3]
```diff
@@ -8,20 +8,34 @@ export const generateSurveySingleUseId = (isEncrypted: boolean): string => {
   if (!isEncrypted) {
     return cuid;
   }
-  if (!FORMBRICKS_ENCRYPTION_KEY) {
-    throw new Error("FORMBRICKS_ENCRYPTION_KEY is not defined");
+
+  if (!ENCRYPTION_KEY) {
```
Comment (mattinannt) [inline apps/web/lib/singleUseSurveys.ts:12]: ENCRYPTION_KEY is required so you can remove this check

File: apps/web/lib/singleUseSurveys.ts [hunk 4]
```diff
@@ -8,20 +8,34 @@ export const generateSurveySingleUseId = (isEncrypted: boolean): string => {
...
+
+  const encryptedCuid = symmetricEncrypt(cuid, ENCRYPTION_KEY);
   return encryptedCuid;
 };
 
 // validate the survey single use id
 export const validateSurveySingleUseId = (surveySingleUseId: string): string | undefined => {
-  if (!FORMBRICKS_ENCRYPTION_KEY) {
-    throw new Error("FORMBRICKS_ENCRYPTION_KEY is not defined");
+  if (!ENCRYPTION_KEY) {
```
Comment (mattinannt) [inline apps/web/lib/singleUseSurveys.ts:22]: ENCRYPTION_KEY is required so you can remove this check

**Review by mattinannt — CHANGES_REQUESTED** [review]
@pandeymangg thanks a lot for the PR :-) Looks good in general, just had a few comments ;-)

File: apps/web/lib/singleUseSurveys.ts [hunk 5]
```diff
@@ -8,20 +8,26 @@ export const generateSurveySingleUseId = (isEncrypted: boolean): string => {
...
 // validate the survey single use id
 export const validateSurveySingleUseId = (surveySingleUseId: string): string | undefined => {
-  if (!FORMBRICKS_ENCRYPTION_KEY) {
-    throw new Error("FORMBRICKS_ENCRYPTION_KEY is not defined");
-  }
   try {
-    const decryptedCuid = decryptAES128(FORMBRICKS_ENCRYPTION_KEY!, surveySingleUseId);
+    let decryptedCuid: string | null = null;
+
+    if (surveySingleUseId.length === 64) {
```
Comment (PratikAwaik) [inline apps/web/lib/singleUseSurveys.ts:21]: @pandeymangg Why did you add a check of length === 64 here? Also why did you use AES128 for 64-length string and symmetric decrypt for string of length other than 64?

**Review by mattinannt — APPROVED** [review]
🙏

**Pull request merged**